graphql-spring-boot-autoconfigure/src/main/java/com/oembedler/moon/graphql/boot/GraphQLJavaToolsAutoConfiguration.java

```diff
@@ -49,29 +55,31 @@ public SchemaParser schemaParser(List<GraphQLResolver<?>> resolvers) throws IOEx
         SchemaParserBuilder builder = dictionary != null ? new SchemaParserBuilder(dictionary) : new SchemaParserBuilder();
 
         Resource[] resources = applicationContext.getResources("classpath*:**/*.graphqls");
-        if(resources.length <= 0) {
+        if (resources.length <= 0) {
             throw new IllegalStateException("No *.graphqls files found on classpath.  Please add a graphql schema to the classpath or add a SchemaParser bean to your application context.");
         }
 
-        for(Resource resource : resources) {
+        for (Resource resource : resources) {
             StringWriter writer = new StringWriter();
             IOUtils.copy(resource.getInputStream(), writer);
             builder.schemaString(writer.toString());
         }
 
-        if(scalars != null) {
+        if (scalars != null) {
             builder.scalars(scalars);
         }
 
-        if(options != null) {
+        if (options != null) {
             builder.options(options);
+        } else if (objectMapperConfigurer != null) {
+            com.coxautodev.graphql.tools.ObjectMapperConfigurer c = (mapper, context) -> objectMapperConfigurer.configure(mapper);
+            builder.options(SchemaParserOptions.newOptions().objectMapperConfigurer(c).build());
```

Comment:
Doesn't this override other options if specified?  I think it would be better to copy all options and add the objectMapperConfigurer from the servlet if none is specified in `SchemaParserOptions`.  Something like:

```java
if(objectMapperConfigurer != null && options.getObjectMapperConfigurer() != null) {
    SchemaParserOptions.newOptions()
        .objectMapperConfigurer(c)
        // add the rest of the options here
}
```

Reply:
No, it should not override the options if you have specified them, because it's in in the else if branch - where no options have been provided.

Reply:
Ah, I missed that